src/main.rs

```diff
@@ -197,11 +201,16 @@ async fn entry() -> Result<()> {
 
     // Load options
     let mut opts = Options::parse_from(args);
-    let cli_overrides = PkgOverride {
+    let cli_overrides = Arc::new(PkgOverride {
         pkg_url: opts.pkg_url.take(),
         pkg_fmt: opts.pkg_fmt.take(),
         bin_dir: opts.bin_dir.take(),
-    };
+    });
+    let crate_names = take(&mut opts.crate_names);
+    let opts = Arc::new(opts);
+
+    // Create jobserver client
+    let jobserver_client = create_jobserver_client()?;
```

Comment:
> It’s recommended to call this function very early in the lifetime of a program before any other file descriptors are opened.

We should probably do that, i.e. call it in `main` rather than `entry`

Reply:
Fixed!